src/main/java/com/photolocos/enterprise/service/PhotoService.java

```diff
@@ -43,17 +52,22 @@ public PhotoDTO savePhoto(PhotoDTO photo, MultipartFile image) {
     }
 
     @Override
-    public Set<PhotoDTO> fetchAll() {
-        return null;
+    @Cacheable("allPhotos")
+    public Set<PhotoDTO> fetchAll() throws Exception {
+        return photoDAO.fetchAll();
     }
 
     @Override
-    public Set<PhotoDTO> fetchPhotoByCity(String city) {
-        return null;
+    @Cacheable(value = "photosByStateAndCity")
+    public Set<PhotoDTO> fetchPhotoByStateAndCity(String state, String city) throws Exception {
+        LocationDTO location = new LocationDTO();
+        location.setState(state);
+        location.setCity(city);
+        return photoDAO.fetchByLocation(location);
     }
 
     @Override
-    public Set<LocationDTO> fetchLocationByCity(String city) {
-        return null;
+    public LocationDTO fetchLocationByStateAndCity(String state, String city) throws Exception {
+        return locationDAO.fetchByStateAndCity(state, city);
     }
 }
```

Comment:
Added caching of photo sets returned from the service layer. All cache is evicted when a new photo is saved.